### ISSUE TITLE
Await or .. ? 

### ISSUE THREAD

**Issue by ignazioc** [malcommac/Hydra#10]
I'm trying to use `await` using the syntax from the readme.

    let city = try await(getCityPromise())

The `getCityPromise` function is declared like this:

    func getCityPromise() -> Promise<String> { ... } 

But the compiler raises an error
> cannot invoke 'await' with an argument list of type '(Promise\<String\>)'

Then I looked at the unit tests, and I saw that a special operator `..` is used instead of calling the `await` function.

Using that operator my code doesn't raise any error anymore. Should the documentation be fixed or I'm missing something?

**Comment by malcommac** [comment]
It's a mistake with non labeled parameters 👎 
I've solved it in https://github.com/malcommac/Hydra/commit/b2323a971d2a600edd34d2d3191ab007964348ee and it will be released with 0.9.2.

Due to the fact we can't provide a behaviour like in C#, in order to avoid block of the current thread (ie. the main thread) I suggest you to do `await` operations in another queue by adding it inside an `async` closure.
That's the example fixed inside the README:

```swift
// With `async` we have just defined a Promise which will be executed in a given
// context (if omitted `background` thread is used) and return an Int value.
let asyncFunc = async({ _ -> Int // you must specify the return of the Promise, here an Int
    // With `await` the async code is resolved in a sync manner
    let loggedUser = try await(loginUser(username,pass))
    // one promise...
    let followersList = try await(getFollowers(loggedUser))
    // after another...
    let countUnfollowed = try await(unfollow(followersList))
    // ... linearly
    // Then our async promise will be resolved with the end value
    return countUnfollowed
}).then({ value in // ... and, like a promise, the value is returned
    print("Unfollowed \(value) users")
})
```

**Comment by malcommac** [comment]
This is how the operator are defined:

With .. the call is allowed to throws in case of Promise's failure.

```swift
/// This define a `..` operator you can use instead of calling `await` func.
/// If you use this operator you need to also use `do/catch` in order to catch exception for rejected promises.
prefix operator ..
public prefix func ..<T> (_ promise: Promise<T>) throws -> T {
	return try awaitContext.await(promise)
}
```

While using `..!` if it fails it will return nil.

```swift
// This define a `..!` operator you can use instead of calling `await`.
// Using this operator if promise is rejected for any reason the result is `nil` and no throws is called.
prefix operator ..!
public prefix func ..!<T> (_ promise: Promise<T>) -> T? {
	do {
		return try awaitContext.await(promise)
	} catch {
		return nil
	}
}
```
